### DISCUSSION TITLE
Request body limit

### DISCUSSION THREAD

**Discussion by bpolaszek** [clue/framework-x#206] [Q&A] [answered]
Hello there! 👋 

We've been using [Freddie](https://github.com/bpolaszek/freddie) in production on an IoT platform for about 2 weeks now and it performs really great! 👍 

We just encountered a few errors so far, where Freddie would unexpectely respond a `400` error when it [doesn't see any topic](https://github.com/bpolaszek/freddie/blob/main/src/Hub/Controller/PublishController.php#L55) into the request body, even though that one is properly provided.

After further investigation it looks like when the payload is a little bigger than usual (which can happen sometimes), `$request->getBody()` returns a sort of empty stream, resulting in a payload parsing error. So I went to [Framework X's documentation](https://framework-x.org/docs/api/request/) and I noticed this into the _Upload_ section:

> Note that HTTP requests are currently limited to 64 KiB. Any uploads above this size will currently show up as an empty request body with no file uploads whatsoever.

I honestly didn't see that coming. Is it a limit I could raise myself, or are we stuck forever to tiny payloads? 

Thanks!
Ben

**Comment by SimonFrings** [comment]
Hey @bpolaszek , thanks for bringing this up 👍

This question has been mentioned a few times by now, you can take a closer look at https://github.com/clue/framework-x/discussions/69 and https://github.com/reactphp/http/issues/412, you may find a quick answer there.

Does this help for the start?

**Reply by bpolaszek** [reply]
Hey @SimonFrings,

Thanks for your quick reply. 

Unfortunately it doesn't help, as raising the limit looks possible when using raw [reactphp/http](https://github.com/reactphp/http/issues/412#issuecomment-886244835), but not with Framework X, which doesn't give us access to how `React\Http\Server` [is instantiated](https://github.com/clue/framework-x/blob/main/src/App.php#L234-L236) 😞

**Reply by SimonFrings** [reply]
Yeah, I know the `64 KiB` get in your way sometimes, we have plans to adapt a certain behavior to Framework X that handles this for you, without any bigger (or any) adjustments on your side. I'm not sure when this feature will find its way into the project as we're currently involved in developing the ReactPHP v3 with some major new features.

If you need some adjustments sooner, you can reach out to me or @clue and we can find a way to give this some more priority, maybe through a consulting session if that's an option for you.

For a short answer you can also take https://github.com/cweagans/composer-patches or similar projects and change the `64 KiB` manually for now.

**Reply by bpolaszek** [reply]
No worries, after discussion with my team, there's actually no emergency into fixing this, as most Freddie's biggest payloads are produced by some deprecated feature on our side. 
I'm mostly concerned about maintaining an open source project which relies on Framework X, for which I can't ask users to use a composer patch, but at least, I'll leave a note in the [README](https://github.com/bpolaszek/freddie/blob/main/README.md#payload-limitations) until some solution comes up.
Thanks!
Ben

**Comment by clue** [comment, marked as answer]
@bpolaszek Just merged #311 which adds experimental support for raising the request body limit of the built-in web server, see https://framework-x.org/docs/best-practices/deployment/#request-body-limit. This will be part of the upcoming `v0.18.0` release :+1:
